todo_app/app.py

```diff
@@ -11,17 +15,72 @@
 
 @app.route('/')
 def index():
-    todoitems = get_items()
+    query_params = {
+        "key": os.environ.get('API_KEY'),
+        "token": os.environ.get('TOKEN_KEY')
+    }  
+    url = f"https://api.trello.com/1/lists/{os.environ.get('TODO_LISTID')}/cards"
+    todoitemsresponse = requests.get(url, params = query_params).json()
+    todoitems = []
+    for card in todoitemsresponse:
+        myitem = Item(card['id'],card['idList'],card['name'])
+        todoitems.append(myitem)
     return render_template('index.html', todoitems = todoitems)
 
 @app.route('/add', methods=['POST'])
 def add():
-    if request.method == 'POST':
-        add_item(title=request.form.get('title'))
-        return redirect("http://127.0.0.1:5000/")
-    else:
-        pass
+    query_params = {
+        "key": os.environ.get('API_KEY'),
+        "token": os.environ.get('TOKEN_KEY'),
+        "idList": os.environ.get('TODO_LISTID'),
+        "name": request.form.get('title')
+    }
+    url = "https://api.trello.com/1/cards"
+    requests.post(url, params=query_params)
+    return redirect(url_for("index"))
     
+@app.route('/complete_item', methods=['POST'])
+def complete_item():
+          
+    url = f"https://api.trello.com/1/cards/{request.form.get('id')}"
+    query_params = {
+        "key": os.environ.get('API_KEY'),
+        "token": os.environ.get('TOKEN_KEY'),
+        "idList": os.environ.get('DONE_LISTID')
+    }  
+    requests.put(url, params=query_params)
+        
+    return redirect(url_for("index"))
 
+
+@app.route('/completeitem/<id>', methods=['GET','POST'])
+def completeitem(id):
+    url = f"https://api.trello.com/1/cards/{id}"
+    query_params = {
+        "key": os.environ.get('API_KEY'),
+        "token": os.environ.get('TOKEN_KEY'),
+        "idList": os.environ.get('DONE_LISTID')
+    }
+
+    requests.put(url, params=query_params)
+    
+    return redirect(url_for("index"))
+  
 if __name__ == '__main__':
     app.run()
+
+class Item:
```

Comment:
It would be good to extract this into its own file, to keep it separate from the route processing logic above